### DISCUSSION TITLE
Webhook Controller Memory Limits

### DISCUSSION THREAD

**Discussion by brandongodby78** [Azure/azure-workload-identity#757] [Q&A]
So we have deployed this feature in a lab cluster and the webhook controller deployment has a memory limit of 30m. The pods start fine, but as soon as we deploy an application pod that is configured to use workload identity the controller crashes. We are able to get it working by raising the memory limit of the deployment and redeploy, and then our application uses workload identity just fine. The problem is that the memory limit is eventually reset by something, and then the controller crashes again. We cannot determine what is resetting that memory limit back to 30 after we increase it.

**Comment by aramase** [comment]
Hello 👋🏻  are you using the [AKS add-on for Workload Identity](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview)? If yes, you can't change the resource limits manually as AKS will override it. If you're installing the WI webhook using helm charts, then resource limits and requests are configurable.

cc @karataliu for the resource limits on AKS add-on

**Reply by brandongodby78** [reply]
Yes, we are using the add-on, and here is the error we are getting when we try to use the pods:

Error creating: Internal error occurred: failed calling webhook "mutation.azure-workload-identity.io": failed to call webhook: Post "[https://azure-wi-webhook-webhook-service.kube-system.svc:443/mutate-v1-pod?timeout=10s":](https://azure-wi-webhook-webhook-service.kube-system.svc/mutate-v1-pod?timeout=10s%22:) EOF

**Reply by brandongodby78** [reply]
We are getting this error because the pods are being terminated, and here is the error:

terminatedReason:Reason: OOMKilled - exit code: 137Started at: 2023-02-17T15:22:39-05:00Finished at: 2023-02-17T15:25:47-05:00

**Reply by aramase** [reply]
@karataliu will need to update the memory limits for AKS and perform the rollout which can be a while. I'll let him comment on it.

In the meantime, you can deploy the webhook from the repo here to help you get unblocked and re-enable the add-on when the resource limits have been increased in AKS?

**Reply by brandongodby78** [reply]
Yeah, this sounds good. I will likely wait for the update on the add-in, as I am just doing the POC to prepare for GA in March. Is there a way to get notified or something I can watch to know when the add-in memory limits have been updated?

**Reply by brandongodby78** [reply]
@aramase Thank you for the very fast responses, much appreciated.

**Reply by karataliu** [reply]
Thanks @brandongodby78 @aramase for the feedback, we'll update AKS addon for this part.

**Reply by brandongodby78** [reply]
@karataliu  Can you give me a timeline or a commit I can follow so I know when this change is made? I would like to continue my testing after the add-in is working ok.

**Reply by karataliu** [reply]
You can expect the change rollout to all global regions in ~2 weeks, around March 10.
